packages/compartment-mapper/src/import-hook.js

```diff
@@ -38,23 +39,31 @@ const has = (haystack, needle) => apply(hasOwnProperty, haystack, [needle]);
 const resolveLocation = (rel, abs) => new URL(rel, abs).toString();
 
 /**
+ * @param {string} specifier
  * @param {Error} error - error to throw on execute
+ * @param {CompartmentSources} packageSources
  * @returns {StaticModuleType}
  */
-const postponeErrorToExecute = error => {
+const deferError = (specifier, error, packageSources) => {
```

Comment:
Looks like only the message part of error is used here, so this could be just `specifier, message, packageSource`.

Reply:
I kept it there in case we decide to also pass the stacktrace.

Reply:
and now it's used to rethrow depending on situation